doma-spring-boot-autoconfigure/src/main/java/org/seasar/doma/boot/event/DomaApplicationListener.java

```diff
@@ -0,0 +1,102 @@
+
+    public DomaApplicationListener(String beanName, Method method, BeanFactory beanFactory) {
+
+        int parameterCount = method.getParameterCount();
+        if (parameterCount < 1) {
+            throw new IllegalArgumentException("Must receive an entity");
+        } else if (parameterCount > 2) {
+            throw new IllegalArgumentException("Too many parameters");
+        }
+
+        Class<?> entityClass = method.getParameterTypes()[0];
+        if (entityClass.isAnnotationPresent(Entity.class) == false) {
+            throw new IllegalArgumentException("First parameter must be entity class");
+        }
+
+        Set<Class<?>> contextClasses = Collections.newSetFromMap(new IdentityHashMap<>());
+        Annotation[] annotations = AnnotationUtils.getAnnotations(method);
+        for (Annotation annotation : annotations) {
+            HandleDomaEvent handleDomaEvent = AnnotationUtils
+                    .findAnnotation(annotation.annotationType(), HandleDomaEvent.class);
+            if (handleDomaEvent != null) {
+                contextClasses.add(handleDomaEvent.contextClass());
+            }
+        }
+
+        if (parameterCount == 2) {
+            if (contextClasses.size() > 1) {
+                throw new IllegalArgumentException(
+                        "To annotate with multi annotations must be only entity parameter");
+            }
+
+            Class<?> contextClass = method.getParameterTypes()[1];
+            if (contextClass != contextClasses.iterator().next()) {
+                throw new IllegalArgumentException(
+                        "Mismatch between annotation and event context");
+            }
+            Type t = method.getGenericParameterTypes()[1];
+            if (t instanceof ParameterizedType) {
+                Type typeArg = ((ParameterizedType) t).getActualTypeArguments()[0];
+                if (typeArg != entityClass) {
+                    throw new IllegalArgumentException(
+                            "Mismatch between entity class and variable bound to event context");
+                }
+            }
+        }
+
+        this.contextClasses = Objects.requireNonNull(contextClasses);
+        this.beanName = Objects.requireNonNull(beanName);
+        this.method = Objects.requireNonNull(method);
+        this.beanFactory = Objects.requireNonNull(beanFactory);
+    }
+
+    @Override
+    public void onApplicationEvent(DomaEvent<?, ?> event) {
+        Object entity = event.getSource();
+        Object context = event.getContext();
+        if (shouldHandle(context.getClass())
+                && entity.getClass() == method.getParameterTypes()[0]) {
+            Object[] args;
+            if (method.getParameterCount() == 1) {
+                args = new Object[] { entity };
+            } else {
+                args = new Object[] { entity, context };
+            }
+            Object target = beanFactory.getBean(beanName);
+            ReflectionUtils.makeAccessible(method);
```

Comment:
Is it OK to call `ReflectionUtils.makeAccessible(method);` only once in the constructor?